pybamm/models/submodels/particle_cracking/base_cracking.py

```diff
@@ -0,0 +1,156 @@
+#
+# Base class for particle cracking models.
+#
+import pybamm
+
+
+class BaseCracking(pybamm.BaseSubModel):
+    """Base class for particle cracking models.
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+    domain : dict, optional
+        Dictionary of either the electrode for "Positive" or "Nagative"
+    **Extends:** :class:`pybamm.BaseSubModel`
+
+    Reference
+    Ai, W., Kraft, L., Sturm, J., Jossen, A., & Wu, B. (2019). Electrochemical
+    Thermal-Mechanical Modelling of Stress Inhomogeneity in Lithium-Ion Pouch
+    Cells. Journal of The Electrochemical Society, 167(1), 013512.
+    """
+
+    def __init__(self, param, domain):
+        self.domain = domain
+        super().__init__(param)
+
+    def _get_standard_variables(self, l_cr):
+        domain = self.domain.lower() + " particle"
+        if self.domain == "Positive":
+            l_cr0 = self.param.l_cr_p_0
+        else:
+            l_cr0 = self.param.l_cr_n_0
+        l_cr_av = pybamm.x_average(l_cr)
+        variables = {
+            self.domain + " particle crack length [m]": l_cr * l_cr0,
+            self.domain + " particle crack length": l_cr,
+            f"X-averaged {domain} crack length": l_cr_av,
+            f"X-averaged {domain} crack length [m]": l_cr_av * l_cr0,
+        }
+        variables.update(self._get_standard_surface_variables(l_cr))
+        return variables
+
+    def _get_mechanical_results(self, variables):
+        c_s_rav = variables[
+            "R-averaged " + self.domain.lower() + " particle concentration"
+        ]
+        c_s_surf = variables[self.domain + " particle surface concentration"]
+        T_xav = variables["X-averaged cell temperature"]
+        c_s_rxav = variables[
+            "Average " + self.domain.lower() + " particle concentration"
+        ]
+
+        if "Cell thickness change [m]" not in variables:
+            cell_thickness_change = (
+                T_xav * self.param.alpha_T_cell_dim
```

Comment:
here you are doing the dimensionless temperature multiplied by the dimensional thermal expansion coefficient. I thin this should either be `(T_dim - T_ref) * alpha_T_cell_dim` or just `T*alpha_T_cell`.

I think it would be more consistent with how other submodels are written in pybamm to have the expression for thickness change, stress etc. written in dimensionless form and then add the dimensional versions into the variables dict at the end. However, given that this works and you have already addressed so many other comments, I'd be be happy to leave this as is for now

Reply:
Thanks for commenting. Here `T_axv` is the X-averaged `T` for temperature rise, because this is a lumped variable for cell thickness change. I can add `T_xav * self.param.Delta_T` where `self.param.Delta_T = 1`. That's why the current implementation works. Fine to make the cell thickness change dimensionless in the following versions.

Reply:
Thanks, sounds good. I was just worried about `Delta_T` getting changed and then introducing an error here.